functional/iak-idevid-register-with-certificates/test.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+HTTP_SERVER_PORT=8080
+AGENT_ID="d432fbb3-d2f1-4a97-9ef7-75bd81c00000"
+TPM2_OPENSSL="https://github.com/tpm2-software/tpm2-openssl/releases/download/1.2.0/tpm2-openssl-1.2.0.tar.gz"
+CA_PWORD="keylimeca"
+CERT_DIR="/var/lib/keylime"
+TPM_CERTS="/var/lib/keylime/tpm_cert_store"
+
+rlJournalStart
+
+    rlPhaseStartSetup "Do the keylime setup"
+        rlRun 'rlImport "./test-helpers"' || rlDie "cannot import keylime-tests/test-helpers library"
+        rlAssertRpm keylime
+        # update /etc/keylime.conf
+        limeBackupConfig
+        # tenant, set to true to verify ek on TPM
+        rlRun "limeUpdateConf agent enable_iak_idevid true"
+        rlRun "limeUpdateConf registrar tpm_identity iak_idevid"
+        # if TPM emulator is present
+        if limeTPMEmulated; then
+            # start tpm emulator
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            rlRun "limeCondStartAbrmd"
+        fi
+    rlPhaseEnd
+
+    rlPhaseStartSetup "Install tpm2-openssl to generate csrs with TPM keys"
```

Comment:
Currently, Fedora does have tpm2-openssl provider packaged, just tpm2-openssl-engine. By any chance, would you know if it is possible to do the setup using tools that currently exist in Fedora (not the engine though as that is deprecated).

Reply:
Unfortunately the engine does not work here, I have tested and had to use as the provider in the end rather than the engine.